ecommerce/frontend/src/components/checkout/EditCardModal.js

```diff
@@ -0,0 +1,157 @@
+import { useState } from 'react'
+import Cards from 'react-credit-cards'
+import 'react-credit-cards/es/styles-compiled.css'
+import { Modal, Form, Input } from 'antd'
+
+export const EditCardModal = ({
+    cardProps = {
+        cvc: '',
+        expiry: '',
+        number: '',
+        name: '',
+    },
+    cardId = {},
+    cardName = '',
+    visible = false,
+    setVisible = () => {},
+}) => {
+    /* This might not remounted after closing and opening again, be careful */
+    const [cvc, setCvc] = useState(cardProps.cvc)
+    const [expiry, setExpiry] = useState(cardProps.expiry)
+    const [name, setName] = useState(cardProps.name)
+    const [number, setNumber] = useState(cardProps.number)
+    const [cardNickname, setCardNickname] = useState(cardName)
```

Comment:
use `form.getFieldsValue()` in a `onFieldsChange` callback from form like I showed you